fix-#107: Added footer

Mohitsen11 opened the pull request:
## Summary

This PR adds footer in the homepage.

## Description

This PR adds footer in the homepage as it was not added as per the figma file. I created a footer-layout.tsx file and import it at the end of the homepage. 

## Images

![Screenshot (21)](https://github.com/krishnaacharyaa/wanderlust/assets/114509045/12d1a8d5-71e8-4910-ad18-394b550cc25d)

## Issue(s) Addressed

#107

## Prerequisites

- [x] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

krishnaacharyaa reviewed (changes requested):
Couple of observations.
1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
2. It is visible only in the home-page, it should be every where in the entire app ideally

Mohitsen11 commented:
> Couple of observations.
> 
> 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> 2. It is visible only in the home-page, it should be every where in the entire app ideally


I have optimized it and made it visible in the entire app. 
Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.

krishnaacharyaa commented on frontend/src/layouts/footer-layout.tsx:
That's nice, I see a new folder created, do we have scope to create header as well? similarly and refactor the code?

Mohitsen11 replied:
Yeah, we can do this, header code is written inside the homepage and that can be segregate out as a new header-layout.tsx. That will make the homepage code look clean a bit and in future, we can make changes in the header as well and we can put it where we want.

krishnaacharyaa replied:
Sure lets' refactor, but lets' not make it wrapper component..

krishnaacharyaa replied:
@Mohitsen11, if you feel it is too much at a shot, we can raise seperator PR for refactoring aswell, I am okay with it

Mohitsen11 replied:
Okay, that would be good. Lets have first footer-layout at parent level.

Mohitsen11 replied:
> That's nice, I see a new folder created, do we have scope to create header as well? similarly and refactor the code?

I segregated out 'header code' from the 'home-page.tsx' and made a new 'header-layout.tsx' file inside layouts folder. 
Imported all the necessary modules inside 'header-layout.tsx' and added this component in the homepage to keep the UI unchanged.

Want to make a PR.

krishnaacharyaa replied:
Please raise it, let me once have a look

krishnaacharyaa reviewed (commented):
Can you kindly look into the comments

Mohitsen11 commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -122,6 +123,7 @@ function AddBlog() {
 
   return (
+    <Footer>
     <div className="min-h-screen cursor-default bg-slate-50 px-6 py-8 dark:bg-dark">
```
Yes, if we follow this approach we have to do this for every page.

Yes, we can have it in a parent level.

krishnaacharyaa replied:
Let's have it at parent level

Mohitsen11 replied:
> Let's have it at parent level

Yeah, sure.

krishnaacharyaa commented:
> > Couple of observations.
> > 
> > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> 
> I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> 
> Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.

Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)

Can we have Footer component at the last in the parent component

Mohitsen11 commented:
> > > Couple of observations.
> > > 
> > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > 
> > 
> > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> 
> Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> 
> Can we have Footer component at the last in the parent component

Yes, we can have.

Mohitsen11 commented:
> > > > Couple of observations.
> > > > 
> > > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > > 
> > > 
> > > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> > 
> > 
> > Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> > Can we have Footer component at the last in the parent component
> 
> Yes, we can have.

Mohitsen11 commented:
> > > > > Couple of observations.
> > > > > 
> > > > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > > > 
> > > > 
> > > > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > > > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> > > 
> > > 
> > > Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> > > Can we have Footer component at the last in the parent component
> > 
> > 
> > Yes, we can have.

I've added footer-layout in the parent level.

Mohitsen11 commented:
> > > > > > Couple of observations.
> > > > > > 
> > > > > > 1. It is not optimized for mobile devices (remember this is tailwind is mobile first framework)
> > > > > > 2. It is visible only in the home-page, it should be every where in the entire app ideally
> > > > > 
> > > > > 
> > > > > I have optimized it and made it visible in the entire app. Plus, I've applied another approach to do it. I've created footer-layout component that can take other components as a 'prop' such as homepage or whatever the page this app require to add footer. This way, whenever we add new pages in the future, we can simply pass them as a prop to the footer layout, and the footer will automatically be added to those pages.
> > > > > Eventully this component acts as a wrapper for other components, allowing additional content to be included inside the footer-layout while displaying the copyright information at the bottom.
> > > > 
> > > > 
> > > > Thats really nice initiative @Mohitsen11, But I don't think that would make sense to wrap everyone inside the Footer component, the developer experience won't be good, as this is not a standard practice, We ideally just put footer component in the parent component... And we wrap this way only the context ideally. And it is repetative, as in any page I want footer, I should manual wrap it with footer, DX won't be good in my opinion, But I really appreciate the idea and initiative :)
> > > > Can we have Footer component at the last in the parent component
> > > 
> > > 
> > > Yes, we can have.
> 
> I've added footer-layout in the parent level.
And added 'key prop' to the categoryPill component to resolve the error.

Error:- Warning: Each child in a list should have a unique "key" prop.

krishnaacharyaa commented on frontend/src/layouts/footer-layout.tsx:
Please optimize for the mobile screen, all the calss without prefix would be for mobile and sm: would be for sm screens and above, you are doing just the reverse kindly change it

Mohitsen11 replied:
I updated the css.

krishnaacharyaa reviewed (changes requested):
Request for mobile screen optimization and please reduce the height of the footer, it is looking a bit too much in desktop and relative for mobile screen even

krishnaacharyaa commented on frontend/src/layouts/footer-layout.tsx:
1. 
<img width="376" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/18bee3bc-7690-42b3-8714-902304a8b638">

We have further scope of optimizing it for mobile screens, Doesn't the font look very big ? Please use 16px , tailwind equivalent for both desktop and mobile.

2.
<img width="721" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/e009ad7a-fa21-4b1a-954c-babc2c6a362b">
Please make sure they are aligned.

3. Please reduce the height of the footer 
4. I have to scroll to see the footer in both details page and add blogs page, ideally I shouldn't have to scroll as there is much space which is empty which can be effectively utilized. You might have to change viewport related classes

Mohitsen11 replied:
I have made some changes to meet desired output. I have some question to ask too but before I want feedback on these changes.

krishnaacharyaa reviewed (changes requested):
Thank you for the quick updates Mohit, sorry I am a bit picky when it comes to UI/UX, hope you don't mind, I have noted my observations can you please look into them, feel free to ask questions, so that we don't have to do rework again.

Mohitsen11 commented on frontend/src/layouts/footer-layout.tsx:
```diff
@@ -0,0 +1,19 @@
+      <footer className="relative left-0 bottom-0 sm:h-[8vh] flex flex-col sm:flex-row items-center justify-between sm:py-3 sm:px-5 md:px-8 lg:px-20 text-white bg-[#181818]">
+          <section className="text-lg">
+                WonderLust
+          </section>
```
Correction done!

krishnaacharyaa commented on frontend/src/layouts/footer-layout.tsx:
```diff
@@ -0,0 +1,17 @@
+
+  return (
+      <footer className="w-full text-center self-end absolute bottom-0 sm:h-[7vh] flex flex-col sm:flex-row items-center justify-between sm:px-4 md:px-[2rem] lg:px-[4rem] text-white bg-[#181818]">
+          <section className="text-[16px]">
```
please don't hard code the data, unless there is no other way to solve it. 
1. w.r.t heights use tailwind equivalent of px-2rem something like px-2 or px-1 or px-16 so on
2. use variable for color and define it in tailwind-config.js

Mohitsen11 replied:
Correction done!

krishnaacharyaa commented on frontend/src/pages/add-blog.tsx:
```diff
@@ -137,7 +137,7 @@ function AddBlog() {
       </div>
-      <div className="flex justify-center ">
+      <div className="flex justify-center mb-[1.5rem]">
         <form onSubmit={handleSubmit} className="md:w-5/6 lg:w-2/3">
```
Please remove hard code, and why are we adding this bottom margin?

Mohitsen11 replied:
Done!

krishnaacharyaa commented on frontend/src/pages/details-page.tsx:
```diff
@@ -56,7 +56,7 @@ export default function DetailsPage() {
           <div>
-            <p className="leading-7 text-light-secondary dark:text-dark-secondary md:text-lg">
+            <p className="leading-7 text-light-secondary dark:text-dark-secondary md:text-lg mb-[1rem]">
               {post.description}
```
Please remove hard code, and why are we adding this bottom margin? though we have added I see in desktop screen the UI is not as per desired code.
<img width="1637" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/536af87a-7189-45db-a437-1ee4d0f600e7">

Mohitsen11 replied:
Done!

krishnaacharyaa commented on frontend/src/pages/home-page.tsx:
```diff
@@ -71,7 +71,7 @@ function HomePage() {
         <h1 className="pb-4 text-xl font-semibold dark:text-dark-primary sm:pb-0">All Posts</h1>
-        <div className="flex flex-wrap">
+        <div className="flex flex-wrap mb-[2.8rem]">
           {posts.length === 0
```
Same here

Mohitsen11 replied:
Done!

krishnaacharyaa reviewed (commented):
Kindly address the comments

Mohitsen11 commented:
> Kindly address the comments

The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.

Mohitsen11 commented:
> > Kindly address the comments
> 
> The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.



> > Kindly address the comments
> 
> The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.

Um UI is still breaking. I just checked.
Try to resolve it.

Mohitsen11 commented:
> > > Kindly address the comments
> > 
> > 
> > The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.
> 
> > > Kindly address the comments
> > 
> > 
> > The UI looks good now with the footer layout. I made changes to ensure the footer layout remains controlled as it takes a minimum of 7% of the viewport height. To make it relative to the content height of other components, I adjusted the minimum height of components like add-blog and details-page to 93% of the viewport height.
> 
> Um UI is still breaking. I just checked. Try to resolve it.

krishnaacharyaa reviewed (commented):
Thank you @Mohitsen11 for your efforts :)
As I see... I still see some inconsistency in the implementation
1. 
<img width="1600" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/7d19747a-a3e0-49ca-a978-6d32b6de269a">
the "All Rights Reserved" text is not aligned to the top blogs. 
2. 
<img width="497" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/262154bf-5307-4b98-af42-f03aa8594390">
Footer should have been properly visible, but because of the logic/hard coded, it isn't 

I don't want to burden you with revisions, may be if you want, I'll do it myself and show the changes later to you, else if you wanna do anyways, kindly test all the corner cases with good quality code(only the required code), for perfection you can test it in your mobile using the private branch i.e
<img width="1026" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/e1712a11-bc98-414a-9bc0-f828ff963807">

Just let me know

Mohitsen11 commented on frontend/src/layouts/footer-layout.tsx:
```diff
@@ -0,0 +1,17 @@
+
+  return (
+      <footer className="mt-auto w-full text-center self-end relative bottom-0 min-h-[7vh] sm:h-[7vh] flex flex-col sm:flex-row items-center justify-between py-3 sm:px-4 md:px-8 lg:px-16 text-white bg-dark-footer-background">
+          <section className="text-base">
```
Correction done!

krishnaacharyaa commented:
@Mohitsen11 can you leverage flex-1 property, that should ideally solve the issue, min-h-screen looks good to me ...
https://stackoverflow.com/a/75042043/13431819 , see if this helps

Mohitsen11 commented:
> @Mohitsen11 can you leverage flex-1 property, that should ideally solve the issue, min-h-screen looks good to me ... https://stackoverflow.com/a/75042043/13431819 , see if this helps

Okay.

krishnaacharyaa commented:
@Mohitsen11  Kindly create a bug/issue about "we have to scroll to find the footer even if there is space for the footer in the description, add blog page in both desktop and mobile"
And attach screenshots 
<img width="372" alt="image" src="https://github.com/krishnaacharyaa/wanderlust/assets/116620586/862aa3a5-1dd9-4b7f-ae1b-0c0de800dd4a">

Once done, share the ticket here, Will merge this PR

Mohitsen11 commented:
> @Mohitsen11 Kindly create a bug/issue about "we have to scroll to find the footer even if there is space for the footer in the description, add blog page in both desktop and mobile" And attach screenshots <img alt="image" width="372" src="https://private-user-images.githubusercontent.com/116620586/318427615-862aa3a5-1dd9-4b7f-ae1b-0c0de800dd4a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE5NzQ4NDAsIm5iZiI6MTcxMTk3NDU0MCwicGF0aCI6Ii8xMTY2MjA1ODYvMzE4NDI3NjE1LTg2MmFhM2E1LTFkZDktNGI3Zi1hZTFiLTBjMGRlODAwZGQ0YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwMVQxMjI5MDBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YjZmNjFhYWU5NDhjZDVmYmEyMzBjYWEzNWQxYTgyZjQzZTMwOTVlN2I4YzIxZWI0NGQzYmE1MTQwNDk4MWI1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.sU2r9xySMpQJRkwFgjJzMR3dgKAFbbI02zO_dFY0HhM">
> 
> Once done, share the ticket here, Will merge this PR

Ticket :- 'https://github.com/krishnaacharyaa/wanderlust/issues/121'

krishnaacharyaa reviewed (approved):
LGTM, Merging PR
